native/index.d.ts

```diff
@@ -0,0 +1,68 @@
+// <reference path="../react/react-native.d.ts" />
+
+declare module "styled-components/native" {
```

Comment:
Declaration files published within packages should be at the top level, and not use the `declare module` syntax. Each `.d.ts` file's top level should align to the exports of the `.js` file.

The reason is that if you had multiple files that wrote out `declare module "foo"`, then TypeScript wouldn't know how to pick the "correct" one. Being a module with a unique path on disk means that importing `"foo"` can be determined by the resolution process.